port/zos390/omrvmem.c

```diff
@@ -1352,9 +1361,18 @@ omrvmem_find_valid_page_size(struct OMRPortLibrary *portLibrary, uintptr_t mode,
 		uintptr_t defaultLargePageSize = 0;
 		uintptr_t defaultLargePageFlags = OMRPORT_VMEM_PAGE_FLAG_NOT_USED;
 
-		/* If the page type is NOT_USED (as with the -Xlp<size> options)
-		 * the legacy behavior is to default to FIXED pages.
-		 */
+		/* If the page type is PREFER_PAGEABLE try to get 1M pageable large pages. Otherwise, use a fixed large page. */
+		if (OMRPORT_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE == validPageFlags && ONE_M == validPageSize) {
```

Comment:
Please add extra brackets, do not rely on order of operations

Reply:
Done. https://github.com/eclipse/omr/pull/4762/commits/65e4d3c847cdd2c81d5d1cd7f68d17e4eaff1ccf